### PR TITLE
Add 64Bits Address Space, Fixs

### LINKED ISSUE #21: Create 64-bit address space version of RetDec IDA plugin
Right now, we have only one plugin version for 32-bit address space (which is itself a 64-bit library, see [explanation](https://www.hex-rays.com/products/ida/7.0/docs/api70_porting_guide.shtml)). It should be possible to also create a 64-bit address space version, which would for now decompile only 32-bit binaries (until we add support for some 64-bit architectures).

### PR DESCRIPTION
**Pull request by Maroc-OS**
Add Support for 64Bits Address.

Allow more Warnings and fix must of them.

Fix Warnings between normal and 64bit version.

Add default install to user dir if IDA_PATH not set.

More...

Possible Issue closes : 

Basic : https://github.com/avast-tl/retdec-idaplugin/issues/11
Close: https://github.com/avast-tl/retdec-idaplugin/issues/21

### CODE REVIEW

**Comment by Maroc-OS** [comment]
Cleaner build system that support all Operating Systems:

To Build 64Bit Version use : 

`cmake .. -DIDA_SDK_DIR=$IDA_SDK_DIR -DEA64=1 && make`

OUTPUT : 

```
Configuring RetDec build for ida pro 7 64bits address space.
-- Configuring done
-- Generating done

....

[ 96%] Building CXX object src/idaplugin/CMakeFiles/idaplugin.dir/idaplugin.cpp.o
[ 98%] Building CXX object src/idaplugin/CMakeFiles/idaplugin.dir/plugin_config.cpp.o
[100%] Linking CXX shared library retdec64.dylib
[100%] Built target idaplugin
```

This will produce one of those : retdec64.dylib, retdec64.lib or retdec64.dll for macOS, Linux or Windows.

You can install now for only current user if IDA PRO PATH is not set.

Finally i hope i did not forgot anything because my version have a lot of changes that am testing and cant be sure if i toke all needed parts. :)

**Comment by Maroc-OS** [comment]
Test and give me back @PeterMatula

**Comment by PeterMatula** [comment]
Yeah, I will look into this, but not right away. First I will try to finish bash to python re-implementation of scripts. Then we will create new releases of both RetDec and IDA plugin, so if this is ok, we will probably also create 64-bit plugins.

**Comment by Maroc-OS** [comment]
Sorry for the delay, yes it's completely okay for me. let's keep this for later. portability is more important 👍

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.6)
+cmake_minimum_required(VERSION 3.7)
```
Comment (s3rvac) [inline CMakeLists.txt]: Is CMake 3.7 really required? I am not sure if we want to raise the minimal required CMake version.

Comment (Maroc-OS) [reply CMakeLists.txt]: not really

Comment (Maroc-OS) [reply CMakeLists.txt]: sorry for the timing i still have some changes in my mac will push that asap

**Comment by PeterMatula** [comment]
I did a few changes here, but then I realized that it would be better to work on #36. So I'm closing this and try to merge #36.

Thanks @Maroc-OS

**Pull request closed without merging**